woocommerce/woocommerce-android
# Improve product selector data sync

#### ⚠️ Dependency - FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2748 *
##### *Before merging this PR make sure the PR in FluxC is merged, then update FluxC version.

This PR changes products fetch operation in Product Selector to partial/mild sync style which does not wipe products table. This improves the performance of local search and fixes issues with the featured products not being added to an order.

Modification of data loading process:
1. Don't wipe the products table on the initial products load request.
2. Instead, display the initial page of existing results from the local db, and side re-fetch the first page in the background. When fetch completes, show the updated product list.
3. Paginate local db observable flow according to the current remote data fetch offset value. On each load-more request, fetch the next page of products and emit the next "page" from the local db. 



Closes: #9200


### Testing instructions

* Make sure you can add the featured product to order (from recent, and popular sections) in case the product is not present on the first page of products.
* Test if pagination works as expected
